org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMElementFormatter.java

```diff
@@ -142,8 +143,11 @@ private int formatStartTagElement(DOMElement element, XMLFormattingConstraints p
 			}
 			case collapse: {
 				// collapse empty element: <example></example> -> <example />
-				if (!element.isSelfClosed() && (end == -1 || element.getEndTagOpenOffset() + 1 < end)) {
-					// Do not collapse if range is does not cover the element
+				if (!element.isSelfClosed() && (end == -1 || element.getEndTagOpenOffset() + 1 < end)
+						&& (!isGrammarAwareFormatting()

```

Comment:
Here the element formatter should not use isGrammarAwareFormatting() (it is the participant), I will write just:

`shouldCollapseEmptyElement(element, formatterDocument.getSharedSettings())`

instead of 

`(!isGrammarAwareFormatting() || shouldCollapseEmptyElement(element, formatterDocument.getSharedSettings()))`